lib/ansible/config/base.yml

```diff
@@ -1691,6 +1691,19 @@ RETRY_FILES_SAVE_PATH:
   ini:
   - {key: retry_files_save_path, section: defaults}
   type: path
+RUN_VARS_PLUGINS:
+  name: When should vars plugins run relative to inventory
+  default: demand
+  description:
+    - This setting can be used to optimize vars_plugin usage depending on user's inventory size and play selection.
+    - Setting to 'demand' will run vars_plugins relative to inventory sources anytime vars are 'demanded' by tasks.
```

Comment:
```suggestion
    - Setting to C(demand) will run vars_plugins relative to inventory sources anytime vars are 'demanded' by tasks.
```

Reply:
Going to let my ci_complete tests finish and then I'll update these. Thank you for re-reviewing!